client/src/_reducers/auth.js

```diff
@@ -0,0 +1,20 @@
+import dayjs from 'dayjs'
+import { ADD_AUTH, REMOVE_AUTH } from '_actions/auth'
+
+const initialAuthState = {
+  token: localStorage.getItem('pie-token'),
+  expiration: localStorage.getItem('pie-expiration') || dayjs().toDate()
+}
+
+export default function auth(state = initialAuthState, action) {
+  switch (action.type) {
+    case ADD_AUTH:
+    case REMOVE_AUTH:
+      return Object.assign({}, state, {
```

Comment:
Instead of `Object.assign` I have been using the spread operator last couple years, which also allows you to create a new object via a merge of current state and updates. in this case it'd look something like this: {...state, token: action.token, expiration }